src/modelAnalysis/FVA/fluxVariability.m

```diff
@@ -197,38 +217,80 @@
             QPproblem.F = spdiags(minNorm,0,nRxns,nRxns);
             %quadratic optimization
             solution = solveCobraQP(QPproblem);
-            if isempty(solution.full)
-                %pause(eps)
-            end
-            Vmax(:,rxnBool)=solution.full(1:nRxns,1);
+            Vmax(:,i)=solution.full(1:nRxns,1);
+        elseif (OutputMatrix && isequal(method,'1-norm'))
+            i
+            vSparse = sparseFBA(LPproblem,'max',0,0,'l1');
+            Vmax(:,i) = vSparse;
+        elseif (OutputMatrix && isequal(method,'0-norm'))
+            vSparse = sparseFBA(LPproblem,'max',0,0);
+            Vmax(:,i) = vSparse;
+        elseif (OutputMatrix && isequal(method,'FBA'))
+            Vmax(:,i) = LPsolution.full;
+        elseif (OutputMatrix && isequal(method,'minOrigSol'))
+            LPproblemMOMA = LPproblem;
+            LPproblemMOMA=rmfield(LPproblemMOMA,'csense');
+            LPproblemMOMA.A = model.S;
+            LPproblemMOMA.S = LPproblemMOMA.A;
+            LPproblemMOMA.b = model.b;
+            LPproblemMOMA.lb(find(model.c)) = objValue;
+            LPproblemMOMA.lb(i) = maxFlux(i);
+            LPproblemMOMA.ub(i) = maxFlux(i);
+            LPproblemMOMA.rxns = model.rxns;
+            momaSolution = linearMOMA(model,LPproblemMOMA);
+            Vmax(:,i)=momaSolution.x;
         end
 
         LPproblem.osense = 1;
         LPsolution = solveCobraLP(LPproblem);
         %take the maximum flux from the flux vector, not from the obj -Ronan
-        minFlux(i) = LPsolution.full(LPproblem.c~=0);
+        if LPsolution.full(LPproblem.c~=0)<LPproblem.lb(i) %takes out tolerance issues 

```

Comment:
you are mixing (i) and (c~=0) here. LPProblem(i) should always be LPProblem(c~=0) since this is for the current reaction. i is only correct, if a full FVA is performed.
i is the index in the resultVector, which corresponds to c~=0 in the LPProblem Vectors.